src/components/Dashboard/ProjectActivity/PaymentActivity.tsx

```diff
@@ -60,98 +59,88 @@ export function PaymentActivity({
     return override ? override : e.note
   }
 
-  useEffect(() => {
-    setLoading(true)
-
-    querySubgraph(
-      {
-        entity: 'payEvent',
-        keys: ['amount', 'beneficiary', 'note', 'timestamp', 'txHash'],
-        first: pageSize,
-        skip: pageNumber * pageSize,
-        orderDirection: 'desc',
-        orderBy: 'timestamp',
-        where: projectId
-          ? {
-              key: 'project',
-              value: projectId.toString(),
-            }
-          : undefined,
-      },
-      res => {
-        if (!res) return
-
-        const newEvents = [...payEvents]
-        newEvents.push(...res.payEvents.map(e => parsePayEventJson(e)))
-        setPayEvents(newEvents)
+  const { data: payEvents } = useSubgraphQuery(
+    {
+      entity: 'payEvent',
+      keys: ['id', 'amount', 'beneficiary', 'note', 'timestamp', 'txHash'],
+      first: pageSize,
+      skip: pageNumber * pageSize,
+      orderDirection: 'desc',
+      orderBy: 'timestamp',
+      where: projectId
+        ? {
+            key: 'project',
+            value: projectId.toString(),
+          }
+        : undefined,
+    },
+    {
+      onSuccess: data => {
         setLoading(false)
-        setCount(newEvents.length)
+        setCount(data?.length)
```

Comment:
something unexpected is happening with the onSuccess callbacks. seems like `setCount` isn't being called every time.

If I tab from Pay to Redeem events then back to Pay, for example, it'll show the `count` of Redeem events under the Pay tab. It might be something to do with the component not being destroyed, so the hook isn't reinitialized.